frontend/components/admin.jsx

```diff
@@ -19,68 +19,86 @@ class Admin extends React.Component {
       socket:socket,
-      solved:false,
-      tableData : [],
-      gameState : [''],
+      puzzleData:[],
+      gameStarted: false
     };
-    this.adminPing();
+    this.adminPing('load');
   }
 
-  adminPing() {
+  adminPing(trigger) {
     const { socket } = this.state;
-    socket.emit('admin_ping', {});
+    socket.emit('admin_ping', { 'trigger':trigger });
   }
 
   startGame() {
-    console.log('this will be the start game button');
+    this.adminPing('start_game');
   }
 
-  renderClients(table) {
-    const { tableData } = this.state;
-    const clientElements = Object.entries(tableData[table]).map((elements,i) => {
-      var clients = elements[1];
-      return (
-        <li key={i}>{clients}</li>
-      );
-    });
-    return clientElements;
+  stopGame() {
+    this.adminPing('stop_game');
   }
 
-  renderPuzzle(table) {
-    const { gameState } = this.state;
-    const puzzleElements = Object.entries(gameState[table]).map(elements => {
-      var puzzle = elements[0];
-      var solved = elements[1]['solved'];
+  renderPuzzle(puzzle) {
+    const { puzzleData } = this.state;
+    var solvedCounter = 0;
+    var tableCounter = puzzleData[puzzle].length;
+    const clientElements = Object.values(puzzleData[puzzle]).map((elements,i) => {
+      var innerElements = Object.entries(elements);
+      var [[table]] = innerElements;
+      var { started } = elements[table];
+      var { solved } = elements[table];
+      var hintCount = elements[table]['hint_count'];
+      if (solved === true) {
+        solvedCounter += 1;
+      }
       return (
-        <li key={puzzle}>puzzle {puzzle}: <CheckIcon solved={solved}/></li>
+        <li key={i}>table: {table} --> started: <CheckIcon solved={started}/> solved: <CheckIcon solved={solved}/> hint count: {hintCount}</li>
       );
     });
-    return puzzleElements;
+    return [clientElements, solvedCounter, tableCounter];
   }
 
   render() {
-    const { tableData } = this.state;
-    const tableElements = Object.entries(tableData).map(elements => {
-      var table = elements[0];
-      var clientList = this.renderClients(table);
-      var tableGameState = this.renderPuzzle(table);
-      return (
-        <div key={table}>
-          <h1>{table}</h1>
-          <p>Participants: {clientList} </p>
-          <p>Status: {tableGameState} </p>
+    const { gameStarted, puzzleData } = this.state;
+    let message;
+    if (gameStarted === false) {
+      message = "You have not started the game yet";
+    } else {
+      message = "Game has been started";
+    }
+    const tableElements = Object.entries(puzzleData).map(elements => {
+      var [puzzle] = elements;
+      if (puzzle === 'start_time') {
+        return;
 
-        </div>
-      );
+      } else {
+        var puzzleRendered = this.renderPuzzle(puzzle);
+        var [clientList, solvedCounter, tableCounter] = puzzleRendered;
+        return (
+          <div key={puzzle}>
+            <h1>Puzzle  {puzzle}:</h1>
```

Comment:
what's the reasoning behind displaying this based on puzzles rather than tables?
if there's a good reason, should we have a button to switch between the two? having it shown by table seems useful to me.

Reply:
Chris wanted it to be displayed by puzzle number. Can work on the switch functionality if needed